### PR TITLE
E_CLASSROOM-258 [Admin] [FE/BE] Add/Edit categories functionality

### PR DESCRIPTION
**Pull request by ErickDoroy**
### Issue Link
- https://framgiaph.backlog.com/view/E_CLASSROOM-258
### Definition of Done
- [x] Implement FE integration for
- [x] Adding category name and description
- [x] Editing category name and description

### Relate PR:
- https://github.com/framgia/sph-classroom-els-be/pull/54
#### Main note
Category page:
- http://localhost:3003/admin/categories?page=1

Click the add button to redirect to this link:
- http://localhost:3003/admin/add-category

Click the edit button to redirect to this link:
- http://localhost:3003/admin/edit-category/:id
#### Pages Affected
- N/A

### Scenarios/ Test Cases
- [ ] Able to add a category
- [ ] Able to edit a category
#### User Interface
- N/A
#### User Experience 
- N/A
#### Functionality
- N/A
### Screenshots
addcategory
editcategory

### CODE REVIEW

File: src/pages/admin/categories/AddEditCategory/index.js [hunk 1]
```diff
@@ -1,18 +1,64 @@
...
+          toast('Success', 'Successfully Added Category.');
+          history.push('/admin/categories');
+        });
+      } else { 
+        toast('Processing', 'Updating Category...');
+        CategoryApi.update(name, description, category_id).then(() => {
+          toast('Success', 'Successfully Updated Category.');
+          history.push('/admin/categories');
+        });
+      }
```
Comment (karlolee50) [inline src/pages/admin/categories/AddEditCategory/index.js]: use else if for the other pathname.

the else should show an error saying something like "invalid url"

File: src/pages/admin/categories/AddEditCategory/index.js [hunk 2]
```diff
@@ -1,18 +1,64 @@
...
+    } catch (error) {
+      toast('Error', 'The given data was invalid.');
+      setSubmitStatus(false);
+      console.log(error);
```
Comment (karlolee50) [inline src/pages/admin/categories/AddEditCategory/index.js]: remove console.log if not used

**Comment by catampongan** [comment]
## Working
- [x] Able to add/edit category

## Comment
- [x] Error should display when adding/editing a category with the same category name

-------------------
`Added Category`

Screenshot 2022-02-22 101910

`Edited Category`

Screenshot 2022-02-22 105229


`Error must display`

test

----------

- Total Test Time: 0.75 hr

**Comment by catampongan** [comment]
## Approved PR #115 
- Able to add category name and description
- Able to edit category name and description

--------------
`Added Category`

Screenshot 2022-02-22 101910

`Edited Category`

Screenshot 2022-02-22 105229
-----------
- Total Test Time: 1 hr
- Total Iterations: 1
- Total Comments: 1

**Review by karlolee50 — APPROVED** [review]
Review time: 15 minutes
Comment/s: 2

**Pull request merged**